README.md
# Mosaic HTML

HTML landing page template designed by Cruip.

This HTML template is shipped with Webpack as build tool, and it's based on a custom version of (Eric Alli's Static Site Boilerplate)[https://staticsiteboilerplate.com/].

* [Getting started](#getting-started)
* [Deploy with Netlify](#deploy-with-netlify)

## Getting started

* First, ensure that node.js & npm are both installed. If not, choose your OS and installation method from [this page](https://nodejs.org/en/download/package-manager/) and follow the instructions.
* Next, use your command line to enter your project directory.
* This template comes with a ready-to-use package file called `package.json`. You just need to run `npm install` to install all of the dependencies into your project.
* When `npm` has finished with the install, run `npm run build` to generate the output files into the `dist` folder.

You're ready to go! The most useful task for rapid development is `npm start`. It will start a new server, open up a browser and watch for any SCSS or JS changes; once it compiles those changes, the browser will automatically inject the changed file(s)!

### Compiles and hot-reloads for development
```
npm start
```

### Compiles and minifies for production
```
npm run build
```

## Weback and support
We are shipping our templates with Webpack to let you quickly get into the development process, but we don't discourage you from using any other build tool. So, please note that any request dealing with Webpack (e.g. extra features, customisations, et cetera) is to be considered out of the support scope.

For more information about what support covers, please see our (FAQs)[https://cruip.com/faq/].

CHANGELOG.md
# CHANGELOG.md

## [3.1.0] - 2021-12-13

- Update Tailwind 3
- Several improvements

## [3.0.0] - 2021-10-20

Add Inbox, Calendar and Onboarding pages
Update dependencies and remove some

## [2.2.0] - 2021-09-09

Add Cart, Cart 2 and Pay pages

## [2.1.0] - 2021-09-06

Add Invoices page

## [2.0.2] - 2021-08-30

Improve colors in settings sidebar and fix dashboard icon

## [2.0.1] - 2021-08-30

Sidebar fix

## [2.0.0] - 2021-08-30

Add more pages and components

## [1.5.0] - 2021-07-09

Add Analytics page and more charts

## [1.4.0] - 2021-07-07

Update TailwindCSS and use JIT

## [1.3.0] - 2021-05-06

Several improvements

## [1.2.0] - 2021-04-20

Several minor changes

## [1.1.0] - 2021-04-04

Update Chart.js to v3
Update more dependencies

## [1.0.0] - 2021-04-01

First release

src/js/components/analytics-card-01.js
/* eslint-disable prefer-destructuring */
// Import Chart.js
import {
  Chart, LineController, LineElement, Filler, PointElement, LinearScale, TimeScale, Tooltip,
} from 'chart.js';
// Import TailwindCSS variables
import resolveConfig from 'tailwindcss/resolveConfig';
import tailwindConfig from '../../../config/tailwind.config';
// Import utilities
import { formatThousands, hexToRGB } from '../utils';

// Tailwind config
const fullConfig = resolveConfig(tailwindConfig);

Chart.register(LineController, LineElement, Filler, PointElement, LinearScale, TimeScale, Tooltip);

// A chart built with Chart.js 3
// https://www.chartjs.org/
const analyticsCard01 = () => {
  const ctx = document.getElementById('analytics-card-01');
  if (!ctx) return;
  // eslint-disable-next-line no-unused-vars
  const chart = new Chart(ctx, {
    type: 'line',
    data: {
      labels: [
        '12-01-2020', '01-01-2021', '02-01-2021',
        '03-01-2021', '04-01-2021', '05-01-2021',
        '06-01-2021', '07-01-2021', '08-01-2021',
        '09-01-2021', '10-01-2021', '11-01-2021',
        '12-01-2021', '01-01-2022', '02-01-2022',
        '03-01-2022', '04-01-2022', '05-01-2022',
        '06-01-2022', '07-01-2022', '08-01-2022',
        '09-01-2022', '10-01-2022', '11-01-2022',
        '12-01-2022', '01-01-2023',
      ],
      datasets: [
        // Indigo line
        {
          label: 'Current',
          data: [
            5000, 8700, 7500, 12000, 11000, 9500, 10500,
            10000, 15000, 9000, 10000, 7000, 22000, 7200,
            9800, 9000, 10000, 8000, 15000, 12000, 11000,
            13000, 11000, 15000, 17000, 18000,
          ],
          fill: true,
          backgroundColor: `rgba(${hexToRGB(fullConfig.theme.colors.blue[500])}, 0.08)`,
          borderColor: fullConfig.theme.colors.indigo[500],
          borderWidth: 2,
          tension: 0,
          pointRadius: 0,
          pointHoverRadius: 3,
          pointBackgroundColor: fullConfig.theme.colors.indigo[500],
        },
        // Gray line
        {
          label: 'Previous',
          data: [
            8000, 5000, 6500, 5000, 6500, 12000, 8000,
            9000, 8000, 8000, 12500, 10000, 10000, 12000,
            11000, 16000, 12000, 10000, 10000, 14000, 9000,
            10000, 15000, 12500, 14000, 11000,
          ],
          borderColor: fullConfig.theme.colors.gray[300],
          fill: false,
          borderWidth: 2,
          tension: 0,
          pointRadius: 0,
          pointHoverRadius: 3,
          pointBackgroundColor: fullConfig.theme.colors.gray[300],
        },
      ],
    },
    options: {
      layout: {
        padding: 20,
      },
      scales: {
        y: {
          beginAtZero: true,
          grid: {
            drawBorder: false,
          },
          ticks: {
            callback: (value) => formatThousands(value),
          },
        },
        x: {
          type: 'time',
          time: {
            parser: 'MM-DD-YYYY',
            unit: 'month',
            displayFormats: {
              month: 'MMM YY',
            },
          },
          grid: {
            display: false,
            drawBorder: false,
          },
          ticks: {
            autoSkipPadding: 48,
            maxRotation: 0,
          },
        },
      },
      plugins: {
        legend: {
          display: false,
        },
        tooltip: {
          callbacks: {
            title: () => false, // Disable tooltip title
            label: (context) => formatThousands(context.parsed.y),
          },
        },
      },
      interaction: {
        intersect: false,
        mode: 'nearest',
      },
      maintainAspectRatio: false,
    },
  });
};

export default analyticsCard01;
